lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -631,7 +632,11 @@ public function computeChangeSet(ClassMetadata $class, $entity)
         $actualData = [];
 
         foreach ($class->reflFields as $name => $refProp) {
-            $value = $refProp->getValue($entity);
+            if (method_exists($refProp, 'isInitialized') && $refProp->getDeclaringClass()->isInstance($entity)) {
```

Comment:
@beberlei Thank you very much for the review, it gave me a lot of missing insight.

The current state of the test for this issue seems to be broken, as the issue does not arise for public properties, because for them an instance of `\Doctrine\Common\Reflection\RuntimePublicReflectionProperty` is created, which is checking the object property with `isset` prior to accessing it's value:
```php
return isset($object->$name) ? parent::getValue($object) : null;
```
So the error happens only for protected and private non-initialized properties without a default value.

I reverted my previous fix and updated the test accordingly, it is again failing.

If I understand correctly, we need to first update doctrine/reflection, then update the dependency in doctrine/persistence and implement the fix there, and then update the dependencies in doctrine/orm and the test will pass.

I created https://github.com/doctrine/reflection/pull/22 in doctrine/reflection to fix some PHP 7.4 issues that currently block me from running the tests, then I will proceed with the implementation.